File: static/js/chart.js
/*
 * @Description: Copyright (c) ydfk. All rights reserved
 * @Author: ydfk
 * @Date: 2025-11-17 18:56:12
 * @LastEditors: ydfk
 * @LastEditTime: 2025-11-17 19:14:26
 */
document.addEventListener("DOMContentLoaded", function () {
  var canvas = document.getElementById("lineChart");
  if (!canvas) {
    return;
  }

  fetch("/api/chart/data")
    .then(function (res) {
      return res.json();
    })
    .then(function (data) {
      var ctx = canvas.getContext("2d");

      new Chart(ctx, {
        type: "line",
        data: {
          labels: data.labels || [],
          datasets: [
            {
              label: data.seriesName || "数据",
              data: data.values || [],
              fill: false,
              borderColor: "rgba(75, 192, 192, 1)",
              borderWidth: 2,
              tension: 0.25,
              pointRadius: 3,
            },
          ],
        },
        options: {
          responsive: true,
          maintainAspectRatio: false,
          interaction: {
            mode: "index",
            intersect: false,
          },
          scales: {
            x: {
              title: {
                display: !!data.xLabel,
                text: data.xLabel || "",
              },
            },
            y: {
              title: {
                display: !!data.yLabel,
                text: data.yLabel || "",
              },
              beginAtZero: true,
            },
          },
          plugins: {
            legend: {
              display: true,
            },
            tooltip: {
              enabled: true,
            },
          },
        },
      });
    })
    .catch(function (err) {
      console.error("加载图表数据失败:", err);
    });
});

File: internal/handler/chart.go
/*
 * @Description: Copyright (c) ydfk. All rights reserved
 * @Author: ydfk
 * @Date: 2025-11-17 18:54:32
 * @LastEditors: ydfk
 * @LastEditTime: 2025-11-17 19:12:15
 */
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

// RegisterChartRoutes 注册折线图相关路由
func RegisterChartRoutes(r *gin.Engine) {
	// 首页和 /chart 指向同一页面
	r.GET("/", ChartPage)
	r.GET("/chart", ChartPage)

	api := r.Group("/api")
	{
		api.GET("/chart/data", ChartData)
	}
}

// ChartPage 页面：渲染折线图 HTML
func ChartPage(c *gin.Context) {
	c.HTML(http.StatusOK, "chart", gin.H{
		"Title": "折线图示例",
	})
}

// ChartData API：返回折线图数据
func ChartData(c *gin.Context) {
	labels := []string{"周一", "周二", "周三", "周四", "周五", "周六", "周日"}
	values := []int{120, 132, 101, 134, 90, 230, 210}

	c.JSON(http.StatusOK, gin.H{
		"labels":     labels,
		"values":     values,
		"seriesName": "访问量",
		"xLabel":     "日期",
		"yLabel":     "访问量",
	})
}
